run.py

```diff
@@ -78,20 +91,75 @@ def cleanup():
     for f in filelist:
         os.remove(f)
 
+def container_name(username):
+    name = "jupyter-" + username
+    return name
+
+def docker_exec(container_name,  arg, user = None, working_dir=None):
+    if(working_dir != None and user != None):
+        cmd = "sudo docker exec -it " + "-u "+ user+ " -w "+ working_dir + " "+ container_name + " python3 run_container.py --test " + arg
```

Comment:
concatenation of strings should be avoided. Or you declare an array of strings and merge it in the end, or you use one string with placeholders (`"%s, %s, %s" % ("val1", "val2", "val3")`)